platforms/avr/drivers/ws2812_bitbang.c

```diff
@@ -172,3 +150,33 @@ static inline void ws2812_sendarray_mask(uint8_t *data, uint16_t datlen, uint8_t
 
     SREG = sreg_prev;
 }
+
+ws2812_led_t ws2812_leds[WS2812_LED_COUNT];
```

Comment:
This is potentially going to cause issue as there are already keyboards that overflow the available ram. Worse that you generally only find out at runtime.

Reply:
RGB Matrix already had an `rgb_led_t` array in the WS2812 shim, and RGBLight currently contains its own internal array which it operates on, and which will be removed when I get to fixing up that side. So I don't think there will be any significant increase in RAM usage, at least not from the addition of this array.

Reply:
> So I don't think there will be any significant increase in RAM usage, at least not from the addition of this array.

So the RGBLight buffer is going to be removed as part of this PR?

Reply:
Yes, I'm only testing with RGB Matrix for now just to verify the driver changes work. Refactoring RGBLight to use the new driver API will be a separate part of this PR.